plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/models/RunningPodAdapterFactory.java

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors: Red Hat, Inc.
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.models;
+
+import org.eclipse.core.runtime.Adapters;
+import org.eclipse.core.runtime.IAdapterFactory;
+import org.jboss.tools.openshift.internal.core.util.ResourceUtils;
+
+import com.openshift.restclient.model.IPod;
+
+/**
+ * This factory adapts ResourceWrapper for a running pod to IRunningPod interface,
+ * in this way allowing to filter running pods for menus.
+ */
+public class RunningPodAdapterFactory implements IAdapterFactory {
+
+	@Override
```

Comment:
I would add `@SuppressWarnings("unchecked")`, because the method ooks really yellowish now